### PR TITLE
Protect kubeproxy deployed via kube-up from system OOMs

### PR DESCRIPTION
**Pull request by vishh**
This change is necessary until it can be moved to Guaranteed QoS Class.

For #40573

### CODE REVIEW

**Comment by justinsb** [comment]
@k8s-bot kops aws e2e test this

File: cluster/saltbase/salt/kube-proxy/kube-proxy.manifest [hunk 1]
```diff
@@ -59,7 +59,7 @@ spec:
     command:
     - /bin/sh
     - -c
-    - kube-proxy {{api_servers_with_port}} {{kubeconfig}} {{cluster_cidr}} --resource-container="" {{params}} 1>>/var/log/kube-proxy.log 2>&1
+    - echo -998 > /proc/$$$/oom_score_adj && kube-proxy {{api_servers_with_port}} {{kubeconfig}} {{cluster_cidr}} --resource-container="" {{params}} 1>>/var/log/kube-proxy.log 2>&1
```
Comment (freehan) [inline cluster/saltbase/salt/kube-proxy/kube-proxy.manifest]: I just tried this. No sure why it failed for me. `$$` works for me tough. But e2e is not complaining. 

```
$ cat  /proc/$$$/oom_score_adj
cat: /proc/29756$/oom_score: No such file or directory
```

Comment (vishh) [inline cluster/saltbase/salt/kube-proxy/kube-proxy.manifest]: Kubelet evaluates the command and that's why there is the extra '$'.

**Comment by vishh** [comment]
@k8s-bot kops aws e2e test this

**Comment by vishh** [comment]
@freehan this PR is critical for v1.6 to avoid regressions in GKE

File: cluster/saltbase/salt/kube-proxy/kube-proxy.manifest (hunk 1 above)
Comment (freehan) [inline cluster/saltbase/salt/kube-proxy/kube-proxy.manifest]: I see.

**Pull request merged**